website/payments/admin.py

```diff
@@ -285,6 +387,126 @@ def queryset(self, request, queryset) -> QuerySet:
         return queryset
 
 
+class PaymentsInline(admin.TabularInline):
+    """The inline for payments in the Batch admin"""
+
+    model = Payment
+    readonly_fields = (
+        "topic",
+        "paid_by",
+        "amount",
+        "created_at",
+        "processing_date",
+        "notes",
+    )
+    form = BatchPaymentInlineAdminForm
+    extra = 0
+    max_num = 0
+    can_delete = False
+
+
+@admin.register(Batch)
+class BatchAdmin(admin.ModelAdmin):
+    """Manage payment batches"""
+
+    inlines = (PaymentsInline,)
+    list_display = (
+        "description",
+        "start_date",
+        "end_date",
+        "total_amount",
+        "payments_count",
+        "processing_date",
+        "processed",
+    )
+    fields = (
+        "description",
+        "processed",
+        "processing_date",
+        "total_amount",
+    )
+    search_fields = ("description",)
+
+    def get_readonly_fields(self, request: HttpRequest, obj: Batch = None):
+        default_fields = (
+            "processed",
+            "processing_date",
+            "total_amount",
+        )
+        if obj and obj.processed:
+            return ("description",) + default_fields
+        return default_fields
+
+    def delete_model(self, request, batch):
+        if batch.processed:
+            messages.error(request, "You cannot delete a processed batch.")
+        else:
+            super().delete_model(request, batch)
+
+    def has_delete_permission(self, request, obj=None):
+        if isinstance(obj, Batch):
+            if obj.processed:
+                return False
+
+        return super().has_delete_permission(request, obj)
+
+    def get_urls(self) -> list:
+        urls = super().get_urls()
+        custom_urls = [
+            path(
+                "<uuid:pk>/process/",
+                self.admin_site.admin_view(admin_views.BatchProcessAdminView.as_view()),
+                name="payments_batch_process",
+            ),
+            path(
+                "<uuid:pk>/export/",
+                self.admin_site.admin_view(admin_views.BatchExportAdminView.as_view()),
+                name="payments_batch_export",
+            ),
+            path(
+                "new_filled/",
+                self.admin_site.admin_view(
+                    admin_views.BatchNewFilledAdminView.as_view()
+                ),
+                name="payments_batch_new_batch_filled",
+            ),
+        ]
+        return custom_urls + urls
+
+    def save_formset(self, request, form, formset, change):
+        instances = formset.save(commit=False)
+
+        for instance in instances:
+            if instance.batch is None or (
+                instance.batch and not instance.batch.processed
+            ):
```

Comment:
```suggestion
            if instance.batch and not instance.batch.processed:
```

why would you set `instance.batch = None` if it is already `None`?